lib/middleware/operation.js

```diff
@@ -32,55 +36,95 @@ function findPropertyOperations ({ types, method, term, resource }) {
   return operations
 }
 
-function factory (api) {
-  return async (req, res, next) => {
-    if (!req.hydra.resource) {
-      return next()
+function mapOperations ({ api, res, method }) {
+  return res.locals.hydra.resources.flatMap((resource) => {
+    let moreOperations
+    if ('property' in resource) {
+      moreOperations = findPropertyOperations({ resource, method, api }).toArray()
+    } else {
+      const types = clownface({ ...api, term: [...resource.types] })
+      moreOperations = findClassOperations(types, method).toArray()
+    }
+
+    if (moreOperations.length) {
+      return moreOperations.map(operation => ({ resource, operation }))
     }
 
+    return [{ resource, operation: null }]
+  })
+}
+
+function findOperation (operationMiddleware) {
+  return async function (req, res, next) {
     const method = req.method === 'HEAD' ? 'GET' : req.method
-    const types = clownface({ ...api, term: [...req.hydra.resource.types] })
 
-    let operations
-    if (req.hydra.term.equals(req.hydra.resource.term)) {
-      // only look for direct operation when the root resource is requested
-      operations = findClassOperations(types, method)
+    let operations = mapOperations({ api: req.hydra.api, res, method })
+      .filter(({ operation }) => operation)
+
+    if (operationMiddleware) {
```

Comment:
The `operationMiddleware` is not a typical express middleware. I think that could be confusing. Last time we discussed it, I tended to use a more focused callback interface. Now that I have again a look at the code and see the existing `middleware.resource`, I would prefer changing it to a normal express middlware. That means wrapping it with `Router` to support all kinds of middlware arguments (plain function and arrays). `operations` could be attached to `req.hydra`.